File: src/com/l2kt/gameserver/network/clientpackets/RequestExMPCCShowPartyMembersInfo.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.World;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.network.serverpackets.ExMPCCShowPartyMemberInfo;

/**
 * Format:(ch) d
 * @author chris_00
 */
public final class RequestExMPCCShowPartyMembersInfo extends L2GameClientPacket
{
	private int _partyLeaderId;
	
	@Override
	protected void readImpl()
	{
		_partyLeaderId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		Player player = World.getInstance().getPlayer(_partyLeaderId);
		if (player != null && player.isInParty())
			activeChar.sendPacket(new ExMPCCShowPartyMemberInfo(player.getParty()));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestExMPCCShowPartyMembersInfo.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.model.World
import com.l2kt.gameserver.network.serverpackets.ExMPCCShowPartyMemberInfo

/**
 * Format:(ch) d
 * @author chris_00
 */
class RequestExMPCCShowPartyMembersInfo : L2GameClientPacket() {
    private var _partyLeaderId: Int = 0

    override fun readImpl() {
        _partyLeaderId = readD()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        val player = World.getPlayer(_partyLeaderId)
        if (player != null && player.isInParty)
            activeChar.sendPacket(ExMPCCShowPartyMemberInfo(player.party!!))
    }
}